app/containers/message/Components/Audio/index.tsx

```diff
@@ -0,0 +1,212 @@
+				try {
+					setPaused(true);
+					currentTime.value = 0;
+				} catch {
+					// do nothing
+				}
+			}
+		}
+	};
+
+	const setPosition = async (time: number) => {
+		await audioPlayer.setPositionAsync(audioUri.current, time);
+	};
+
+	const getUrl = () => {
+		let url = file.audio_url;
+		if (url && !url.startsWith('http')) {
+			url = `${baseUrl}${file.audio_url}`;
+		}
+		return url;
+	};
+
+	const togglePlayPause = async () => {
+		try {
+			if (!paused) {
+				await audioPlayer.pauseAudio(audioUri.current);
+			} else {
+				await audioPlayer.playAudio(audioUri.current);
+			}
+		} catch {
+			// Do nothing
+		}
+	};
+
+	const onChangeRate = async (value = 1.0) => {
+		await audioPlayer.setRateAsync(audioUri.current, value);
+	};
+
+	const handleDownload = async () => {
+		setLoading(true);
+		try {
+			const url = getUrl();
+			if (url) {
+				const audio = await downloadMediaFile({
+					downloadUrl: `${url}?rc_uid=${user.id}&rc_token=${user.token}`,
+					type: 'audio',
+					mimeType: file.audio_type
+				});
+				await loadAudio(audio);
+				setLoading(false);
+				setCached(true);
+			}
+		} catch {
+			setLoading(false);
+			setCached(false);
+		}
+	};
+
+	const handleAutoDownload = async () => {
+		const url = getUrl();
+		try {
+			if (url) {
+				const isCurrentUserAuthor = author?._id === user.id;
+				const isAutoDownloadEnabled = fetchAutoDownloadEnabled('audioPreferenceDownload');
+				if (isAutoDownloadEnabled || isCurrentUserAuthor) {
+					await handleDownload();
+					return;
+				}
+				setLoading(false);
+				setCached(false);
+			}
+		} catch {
+			// Do nothing
+		}
+	};
+
+	const onPress = () => {
+		if (loading) {
+			return;
+		}
+		if (cached) {
+			togglePlayPause();
+			return;
+		}
+		handleDownload();
+	};
+
+	useEffect(() => {
+		const handleCache = async () => {
+			const cachedAudioResult = await getMediaCache({
+				type: 'audio',
+				mimeType: file.audio_type,
+				urlToCache: getUrl()
+			});
+			if (cachedAudioResult?.exists) {
+				await loadAudio(cachedAudioResult.uri);
+				setLoading(false);
+				setCached(true);
+				return;
+			}
+			if (isReply) {
+				setLoading(false);
+				return;
+			}
+			await handleAutoDownload();
+		};
+		handleCache();
+	}, []);
+
+	useEffect(() => {
+		if (paused) {
+			deactivateKeepAwake();
+		} else {
+			activateKeepAwake();
+		}
+	}, [paused]);
+
+	if (!baseUrl) {
+		return null;
+	}
+	return (
+		<>
+			<Markdown msg={msg} style={[isReply && style]} username={user.username} getCustomEmoji={getCustomEmoji} />
+			<View style={[styles.audioContainer, { backgroundColor: colors.surfaceTint, borderColor: colors.strokeExtraLight }]}>
+				<PlayButton disabled={isReply} loading={loading} paused={paused} cached={cached} onPress={onPress} />
+				<Slider currentTime={currentTime} duration={duration} loaded={!isReply && cached} onChangeTime={setPosition} />
+				<AudioRate onChange={onChangeRate} loaded={!isReply && cached} rate={rate} />
+			</View>
```

Comment:
Can you export this to be used outside `message`?
We're going to need it on new message composer.